### PR TITLE
Migrate to new material buttons

### PR DESCRIPTION
**Pull request by kenzieschmoll**
See flutter.dev/go/material-button-migration-guide

No functional changes - just migrating away from deprecated classes

### CODE REVIEW

**Review by HansMuller — APPROVED** [review]
NICE
LGTM

**Review by jacob314 — APPROVED** [review]
lgtm

File: packages/devtools_app/lib/src/common_widgets.dart [hunk 1]
```diff
@@ -653,25 +653,25 @@ class FilterButton extends StatelessWidget {
 
   @override
   Widget build(BuildContext context) {
-    final colorScheme = Theme.of(context).colorScheme;
+    final theme = Theme.of(context);
     return RoundedOutlinedBorder(
       child: SizedBox(
-        height: defaultButtonHeight,
+        height: theme.buttonTheme.height,
```
Comment (HansMuller) [inline packages/devtools_app/lib/src/common_widgets.dart:659]: Please don't use ButtonTheme, it's obsolete. The component defaults are documented but are not readily available via the widgets or themes. The default height for TextButton is 64 per minimumSize in https://api.flutter.dev/flutter/material/TextButton/defaultStyleOf.html.

File: packages/devtools_app/lib/src/theme.dart [hunk 2]
```diff
@@ -25,45 +25,62 @@ ThemeData themeFor({
...
     accentColor: devtoolsBlue[400],
     backgroundColor: devtoolsGrey[600],
-    canvasColor: background,
     toggleableActiveColor: devtoolsBlue[400],
-    selectedRowColor: devtoolsBlue[600],
-    buttonTheme: theme.buttonTheme.copyWith(minWidth: buttonMinWidth),
+    canvasColor: background,
     scaffoldBackgroundColor: background,
     colorScheme: theme.colorScheme.copyWith(background: background),
+    buttonTheme: theme.buttonTheme.copyWith(
```
Comment (HansMuller) [inline packages/devtools_app/lib/src/theme.dart:70]: Specifying the buttonTheme shouldn't be needed. To change the minimum size of TextButtons, specify minimumSize for TextButtonTheme below.

File: packages/devtools_app/lib/src/common_widgets.dart (hunk 1 above)
Comment (kenzieschmoll) [reply packages/devtools_app/lib/src/common_widgets.dart]: removed all uses of buttonTheme

File: packages/devtools_app/lib/src/theme.dart (hunk 2 above)
Comment (kenzieschmoll) [reply packages/devtools_app/lib/src/theme.dart]: done. Is there a way to set the default button height? I only saw a way to set the min, so had to enforce button height by wrapping certain buttons in SizedBoxes throughout devtools

Comment (HansMuller) [reply packages/devtools_app/lib/src/theme.dart]: You're right, currently there's only support for specifying a minimumSize. Using SizedBox is OK for forcing a particular height.  If you think we should add a fixedSize ButtonStyle parameter to ButtonStyle and the button themes, please file an issue.  It seems reasonable to support that, and it's true that we used to.

Comment (kenzieschmoll) [reply packages/devtools_app/lib/src/theme.dart]: filed https://github.com/flutter/flutter/issues/73741

**Pull request merged**